rooms.js

```diff
@@ -1242,6 +1247,20 @@ class ChatRoom extends BasicRoom {
 		this.chatRoomData = (options.isPersonal ? null : options);
 		Object.assign(this, options);
 		if (this.auth) Object.setPrototypeOf(this.auth, null);
+		/** @type {Room?} */
+		this.parent = null;
+		if (this.parent) {
+			/** @type {Room} */
+			let main = Rooms(this.parent);
+
+			if (main && main.subRooms) {
+				main.subRooms.set(this.id, this);
+				this.parent = main;
+			}
+		}
+
+		/** @type {Map<string, ChatRoom>} */
```

Comment:
Should not be necessary; mark with `// @ts-ignore TypeScript bug: inheritance` if TypeScript is complaining.

Reply:
It is necessary otherwise it flags an error on line 1256.

Reply:
What's the error?